src/features/parser/root/parseRootPageContent.ts

```diff
@@ -0,0 +1,86 @@
+export interface ModuleContent {
+    code: string;
+    url: string;
+}
+
+/**
+ * Main function that orchestrates the parsing of the root page.
+ * It delegates the extraction of taught students' links and modules to other functions.
+ * @param {HTMLElement} content - The content of the root page to parse.
+ * @returns {object} An object containing arrays of ModuleContent for modules and taught students.
+ */
+export function parseRootPage(content: HTMLElement): {
+    modules: ModuleContent[],
+    taught_students: ModuleContent[]
+} {
+    const taught_students = extractTaughtStudentsLinks(content);
+    const modules = extractModulesLinks(content);
+
+    return {
+        modules: modules,
+        taught_students: taught_students
+    };
+}
+
+/**
+ * Extracts the links related to taught students from the specific <p> element.
+ * @param {HTMLElement} content - The content of the page to parse.
+ * @returns {ModuleContent[]} An array of ModuleContent objects representing the links and their labels.
+ */
+export function extractTaughtStudentsLinks(content: HTMLElement): ModuleContent[] {
+    const modules: ModuleContent[] = [];
+
+    const linkContainer = content.querySelector('p a[href*="studres.cs.st-andrews.ac.uk/Teaching/"]')?.parentElement;
```

Comment:
this seems a bit too hard coded, but should be fine for now

I don't think its important, but in future, there might be links that go to not `studres.cs.st-andrews.ac.uk`, which will not be detected by this. So if you can, refactor to get them some other way